### PR TITLE
Document the tasks' inputs and outputs

### PR DESCRIPTION
**Pull request by denisbedin**
https://docs.microej.com/en/feature-microej-sdk-6-tasks-inputs-outputs/SDK6UserGuide/moduleNatures.html

### CODE REVIEW

File: SDK6UserGuide/moduleNatures.rst [hunk 1]
```diff
@@ -109,6 +109,14 @@ adp
 
 **Description**: Executes the Addon Processors.
 
+**Inputs**:
+
+- The project source directory
```
Comment (tdelhomenie-microej) [inline SDK6UserGuide/moduleNatures.rst:114]:
```suggestion
- The project directory
```
"source" seems to refer to the "src" folder.

File: SDK6UserGuide/moduleNatures.rst [hunk 2]
```diff
@@ -119,10 +127,19 @@ This task is used by the following module natures:
...
 loadVeePort
-^^^^^^^^^^^^
+^^^^^^^^^^^
 
 **Description**: Loads the VEE Port.
 
+**Inputs**:
+
+- The path of the VEE Port dropins directory
+- The project which can contain a VEE Port dependency
```
Comment (tdelhomenie-microej) [inline SDK6UserGuide/moduleNatures.rst:137]: The input is a list of files or folders of VEE Port.

File: SDK6UserGuide/moduleNatures.rst [hunk 3]
```diff
@@ -137,6 +154,17 @@ loadApplicationConfiguration
 
 **Description**: Loads the configuration for the Application to execute.
 
+**Inputs**:
+
+- The VEE Port
```
Comment (tdelhomenie-microej) [inline SDK6UserGuide/moduleNatures.rst:159]: Should be "the extracted VEE Port folder", no ? (same for other occurrences)

File: SDK6UserGuide/moduleNatures.rst [hunk 4]
```diff
@@ -137,6 +154,17 @@ loadApplicationConfiguration
...
+- The project classpath which contains the MicroEJ dependent application classes and resources
+- The Full Qualified Name of the Application main class
+- The folder containing the application configuration (``configuration``)
```
Comment (tdelhomenie-microej) [inline SDK6UserGuide/moduleNatures.rst:162]: The debug mode and port, and the system properties, are also inputs.

File: SDK6UserGuide/moduleNatures.rst [hunk 5]
```diff
@@ -258,7 +315,18 @@ This task is used by the following module natures:
...
+
+**Inputs**:
+
+- The VEE Port
+- The configuration file with all the properties set to launch the build of the Executable (``build/properties/target.properties``)
+- The project build classpath
+
+**Outputs**:
+
+- The directory in which the Executable file and the build files have been generated (``build/executable/application``)
```
Comment (tdelhomenie-microej) [inline SDK6UserGuide/moduleNatures.rst:328]:
```suggestion
- The directory in which the Executable file and the build files are generated (``build/executable/application``)
```

File: SDK6UserGuide/moduleNatures.rst [hunk 6]
```diff
@@ -258,7 +315,18 @@ This task is used by the following module natures:
...
+- The Zip file containing the build files generated during the build of the Executable (``build/executable/buildFiles.zip``)
```
Comment (tdelhomenie-microej) [inline SDK6UserGuide/moduleNatures.rst:329]:
```suggestion
- The Zip file containing the generated build files (``build/executable/buildFiles.zip``)
```

File: SDK6UserGuide/moduleNatures.rst [hunk 7]
```diff
@@ -273,6 +341,20 @@ buildWPK
...
 **Description**: Builds the WPK of the Application.
 
+**Inputs**:
+
+- The Application name
+- The Application version
+- The Full Qualified Name of the Application main class
+- The Application JAR file
+- The Application Javadoc
+- The Application classpath
```
Comment (tdelhomenie-microej) [inline SDK6UserGuide/moduleNatures.rst:351]: Its is actually the jars files of the application classpath.

File: SDK6UserGuide/moduleNatures.rst [hunk 8]
```diff
@@ -284,6 +366,18 @@ This task is used by the following module natures:
 buildVirtualDevice
 ^^^^^^^^^^^^^^^^^^
 
+**Inputs**:
+
+- The VEE Port
+- The WPK of the Application
+- The project build classpath
+- The WPK of the Applications that must be pre-installed in the Virtual Device
+- The default tools that must be embedded in the Virtual Device
```
Comment (tdelhomenie-microej) [inline SDK6UserGuide/moduleNatures.rst:375]:
```suggestion
- Libraries that must be embedded in the Virtual Device
```
I think it should not be set as input actually, it is an internal implementation detail. If you are ok, can you create a ticket to remove it please?

File: SDK6UserGuide/moduleNatures.rst [hunk 9]
```diff
@@ -299,6 +393,15 @@ loadKernelExecutable
 
 **Description**: Loads the Kernel Executable file.
 
+**Inputs**:
+
+- The path of the Kernel dropins directory
+- The project which can contain a Kernel dependency
```
Comment (tdelhomenie-microej) [inline SDK6UserGuide/moduleNatures.rst:399]: It is not a project here.

File: SDK6UserGuide/moduleNatures.rst [hunk 10]
```diff
@@ -312,6 +415,17 @@ loadFeatureConfiguration
 
 **Description**: Loads the configuration to build the Feature file of an Application.
 
+**Inputs**:
+
+- The Kernel Virtual Device 
+- The folder in which the Kernel Executable file is copied (``build/kernelExecutable``)
```
Comment (tdelhomenie-microej) [inline SDK6UserGuide/moduleNatures.rst:421]:
```suggestion
- The folder containing the Kernel Executable file (``build/kernelExecutable``)
```

File: SDK6UserGuide/moduleNatures.rst [hunk 11]
```diff
@@ -325,6 +439,17 @@ buildFeature
 
 **Description**: Build the Feature file of an Application.
 
+**Inputs**:
+
+- The Kernel Virtual Device 
+- The folder in which the Kernel Executable file is copied (``build/kernelExecutable``)
```
Comment (tdelhomenie-microej) [inline SDK6UserGuide/moduleNatures.rst:445]:
```suggestion
- The folder containing the Kernel Executable file (``build/kernelExecutable``)
```

File: SDK6UserGuide/moduleNatures.rst [hunk 12]
```diff
@@ -338,6 +463,12 @@ runOnDevice
 
 **Description**: Runs the Executable on a Device.
 
+**Inputs**:
+
+- The VEE Port
+- The directory in which the Executable file has been generated (``build/executable/application``)
```
Comment (tdelhomenie-microej) [inline SDK6UserGuide/moduleNatures.rst:469]:
```suggestion
- The folder containing the Executable file (``build/executable/application``)
```

**Review by tdelhomenie-microej — CHANGES_REQUESTED** [review]

File: SDK6UserGuide/moduleNatures.rst (hunk 4 above)
Comment (denisbedin) [reply SDK6UserGuide/moduleNatures.rst:161]: Done.

File: SDK6UserGuide/moduleNatures.rst (hunk 2 above)
Comment (denisbedin) [reply SDK6UserGuide/moduleNatures.rst]: Exact, done

File: SDK6UserGuide/moduleNatures.rst (hunk 3 above)
Comment (denisbedin) [reply SDK6UserGuide/moduleNatures.rst]: Done.

File: SDK6UserGuide/moduleNatures.rst (hunk 7 above)
Comment (denisbedin) [reply SDK6UserGuide/moduleNatures.rst]: Done.

File: SDK6UserGuide/moduleNatures.rst (hunk 8 above)
Comment (denisbedin) [reply SDK6UserGuide/moduleNatures.rst]: I removed it, It is not an input, it is already annotated with `@internal ` in the plugin https://gitlab.cross/M0090_IDE/M0090_Gradle/-/blob/develop/plugins/src/main/java/com/microej/gradle/tasks/BuildVirtualDeviceTask.java#L77

File: SDK6UserGuide/moduleNatures.rst (hunk 9 above)
Comment (denisbedin) [reply SDK6UserGuide/moduleNatures.rst]: Yes, I changed the inputs to "The list of Kernel Executable files"

**Review by tdelhomenie-microej — APPROVED** [review]

**Pull request merged**